### PR TITLE
Add deleting of old cached dependencies

### PR DESCRIPTION
**Pull request by markbastiaans**
This PR closed internal issue DEP-2887 and adds deleting of old dependencies out of the build cache. This should free up staging space for apps that are upgraded during their lifetimes (e.g. runtime and other dependencies).

### CODE REVIEW

File: tests/requirements.txt [hunk 1]
```diff
@@ -1,6 +1,7 @@
 backoff==1.10.0
 certifi==2019.3.9
 chardet==3.0.4
+cryptography==2.8
```
Comment (cwesdorp) [inline tests/requirements.txt:4]: Why is this added?

Comment (markbastiaans) [reply tests/requirements.txt:4]: This is a test requirement and if you don't specify this, Python 3.4 will break.
